### PR TITLE
Interaction: add `entitiesIn` to line plot

### PR DESCRIPTION
**Pull request by CalvinFernandez**
a component of #3122

### CODE REVIEW

File: src/plots/linePlot.ts [hunk 1]
```diff
@@ -348,6 +348,44 @@ namespace Plottable.Plots {
...
+        dataXRange = { min: bounds.topLeft.x, max: bounds.bottomRight.x };
+        dataYRange = { min: bounds.topLeft.y, max: bounds.bottomRight.y };
+      } else {
+        dataXRange = (<Range> xRangeOrBounds);
+        dataYRange = yRange;
+      }
+
+      const xProjector = Plot._scaledAccessor(this.x());
+      const yProjector = Plot._scaledAccessor(this.y());
+      return this.entities().filter((entity) => {
```
Comment (themadcreator) [inline src/plots/linePlot.ts:379]: perf note. I noticed that plot's `.entities` method does a couple linear time passes on the data. Notably, it uses `_lightweightEntities` then maps them to "heavyweight" entities, which as far as i can tell, just adds a d3 selection. For perf, we should maybe consider using lightweight entities when filtering, then convert to `PlotEntities` as a post process. Lots of other methods use `this.entities()` so, probably not worth in this PR, but for the future.

Comment (themadcreator) [reply src/plots/linePlot.ts:379]: Filed as #3145

File: test/plots/linePlotTests.ts [hunk 2]
```diff
@@ -199,6 +199,60 @@ describe("Plots", () => {
         linePlot.renderTo(svg);
       });
 
+      afterEach(() => {
+        svg.remove();
```
Comment (themadcreator) [inline test/plots/linePlotTests.ts:203]: heh, no brainer

**Review by themadcreator — APPROVED** [review]
Looks good. Keep em coming'

File: src/plots/linePlot.ts (hunk 1 above)
Comment (CalvinFernandez) [reply src/plots/linePlot.ts:379]: 👍

File: src/plots/linePlot.ts [hunk 3]
```diff
@@ -348,6 +348,44 @@ namespace Plottable.Plots {
...
+    public entitiesIn(bounds: Bounds): PlotEntity[];
+    /**
+     * Gets the Entities that intersect the area defined by the ranges.
+     *
+     * @param {Range} xRange
+     * @param {Range} yRange
+     * @returns {PlotEntity[]}
+     */
+    public entitiesIn(xRange: Range, yRange: Range): PlotEntity[];
+    public entitiesIn(xRangeOrBounds: Range | Bounds, yRange?: Range): PlotEntity[] {
```
Comment (hellochar) [inline src/plots/linePlot.ts:365]: nit: should we declare an abstract `entitiesIn` method on XYPlot?

File: src/plots/linePlot.ts [hunk 4]
```diff
@@ -348,6 +348,44 @@ namespace Plottable.Plots {
...
+      let dataXRange: Range;
```
Comment (hellochar) [inline src/plots/linePlot.ts:366]: nit: we use this Range type to specify a range of numbers, but I've also seen `[number, number]` in some places

**Pull request merged**